### PR TITLE
Add initial MMCM feature and PIP support.

### PR DESCRIPTION
**Pull request by litghost**
I believe this solves the remaining unknown bits found in https://github.com/SymbiFlow/symbiflow-arch-defs/issues/1536

### CODE REVIEW

File: fuzzers/031-cmt-mmcm/write_mmcm_reg.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+import argparse
+
+REGISTER_LAYOUT = {
+    'CLKOUT1': [
+        ('LOW_TIME', 6),
+        ('HIGH_TIME', 6),
+        ('OUTPUT_ENABLE', 1),
+        ('PHASE_MUX', 3),
+    ],
+    'CLKOUT2': [
```
Comment (mkurc-ant) [inline fuzzers/031-cmt-mmcm/write_mmcm_reg.py:20]: This layout differs across clock outputs. CLKOUT[0], CLKOUT[5] and CLKFBOUT have different bit fields related to fractional divider.

File: fuzzers/034b-cmt-mmcm-pips/fixup_and_group.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+#
+# Copyright (C) 2017-2020  The Project X-Ray Authors.
+#
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
+#
+# SPDX-License-Identifier: ISC
+"""
+This is a db fixup script that does two things:
```
Comment (mkurc-ant) [inline fuzzers/034b-cmt-mmcm-pips/fixup_and_group.py:12]: You may remove the bit grouping functionality from this script. It has been incorporated into `dbfixup.py`

**Review by mkurc-ant — CHANGES_REQUESTED** [review]
Some changes requested. Otherwise LGTM

File: fuzzers/034b-cmt-mmcm-pips/fixup_and_group.py (hunk 2 above)
Comment (litghost) [reply fuzzers/034b-cmt-mmcm-pips/fixup_and_group.py:12]: https://github.com/SymbiFlow/prjxray/pull/1447/files/b81c84d9d0c3648dd46134d379ec4e2677e52b57#diff-66a33010bffb03af1e370edfd5199582R14-R15

The IN_USE stuff is specific to the MMCM and PLL fixup, so it isn't fully encompased by `dbfixup.py`.

File: fuzzers/031-cmt-mmcm/write_mmcm_reg.py (hunk 1 above)
Comment (litghost) [reply fuzzers/031-cmt-mmcm/write_mmcm_reg.py:20]: Good catch, fixed.

**Review by mkurc-ant — APPROVED** [review]
Ok, Looks good.

**Comment by litghost** [comment]
Zynq fuzzer seems to be getting stuck on `034b-cmt-mmcm-pips`, investigating now.

**Comment by litghost** [comment]
Failure on Zynq is because the following pips never get solved:

```
CMT_TOP_L_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKFBIN.CMT_L_LOWER_B_CLK_FREQ_BB2                                                                                                                                                                                                                  
CMT_TOP_L_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKFBIN.CMT_L_LOWER_B_CLK_FREQ_BB3                                                                                                                                                                                                                  
CMT_TOP_L_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN1.CMT_L_LOWER_B_CLK_FREQ_BB2                                                                                                                                                                                                                   
CMT_TOP_L_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN1.CMT_L_LOWER_B_CLK_FREQ_BB3                                                                                                                                                                                                                   
CMT_TOP_L_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN2.CMT_L_LOWER_B_CLK_FREQ_BB2                                                                                                                                                                                                                   
CMT_TOP_L_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN2.CMT_L_LOWER_B_CLK_FREQ_BB3                                                                                                                                                                                                                   
CMT_TOP_R_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKFBIN.CMT_R_LOWER_B_CLK_FREQ_BB2                                                                                                                                                                                                                  
CMT_TOP_R_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKFBIN.CMT_R_LOWER_B_CLK_FREQ_BB3                                                                                                                                                                                                                  
CMT_TOP_R_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN1.CMT_R_LOWER_B_CLK_FREQ_BB2                                                                                                                                                                                                                   
CMT_TOP_R_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN1.CMT_R_LOWER_B_CLK_FREQ_BB3                                                                                                                                                                                                                   
CMT_TOP_R_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN2.CMT_R_LOWER_B_CLK_FREQ_BB2                                                                                                                                                                                                                   
CMT_TOP_R_LOWER_B.CMT_LR_LOWER_B_MMCM_CLKIN2.CMT_R_LOWER_B_CLK_FREQ_BB3 
```

I've replicated locally, and debugging now.

**Comment by litghost** [comment]
> I've replicated locally, and debugging now.

Zynq is running to completion now, kicking off new build.  Testing with other parts now.

**Pull request merged**